torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -72,13 +74,18 @@ def parallelize_llama(
             )
         apply_compile(model)
 
-    if parallel_dims.dp_enabled:
+    if parallel_dims.dp_enabled or parallel_dims.cp_enabled:
         if parallel_dims.dp_shard_enabled:
             if parallel_dims.dp_replicate_enabled:
                 dp_mesh = world_mesh["dp_replicate", "dp_shard"]
             else:
                 dp_mesh = world_mesh["dp"]
 
+            if parallel_dims.cp_enabled:
+                dp_dim_names = dp_mesh.mesh_dim_names
+                assert isinstance(dp_dim_names, Tuple)
+                dp_mesh = world_mesh[(*dp_dim_names, "cp")]._flatten()
```

Comment:
I remember we want to initialize all the PG in the very beginning. Can we move this to `parallel_dims.py` and use `mesh_dim_name` to rename it to `dp_shard_cp`?